falcon/media/multipart.py

```diff
@@ -610,4 +610,4 @@ def __init__(self) -> None:
         self.max_body_part_buffer_size = 1024 * 1024
         self.max_body_part_count = 64
         self.max_body_part_headers_size = 8192
-        self.media_handlers = self._DEFAULT_HANDLERS
+        self.media_handlers = self._DEFAULT_HANDLERS.copy()
```

Comment:
I think a better course of action might be making these `_DEFAULT_HANDLERS` a tuple of key/value pairs, and then initialize `self.media_handlers = Handlers(self._DEFAULT_HANDLERS)`, then we would make sure that we cannot modify them.

Reply:
(Ah, but that might cause even more issues with circular imports...)

Reply:
OK, I found that your issues do stem from this `.copy()`, we cannot copy `media.Handlers` like that because it doesn't set up all the internals properly.

Or at least, we couldn't before. I went on to implement a propery `.copy()` method thinking it might bite the user trying to copy Handlers in some other scenario too.